.github/workflows/validate-pr.yml

```diff
@@ -0,0 +1,20 @@
+name: Validate PRs
+
+on:
+  pull_request_target:
+    types:
+      - opened
+      - reopened
+      - edited
+
+permissions:
+  pull-requests: read
+
+jobs:
+  validate-pr-title:
+    name: Validate PR title
+    runs-on: ubuntu-latest
+    steps:
+      - uses: amannn/action-semantic-pull-request@c3cd5d1ea3580753008872425915e343e351ab54  # v5.2.0
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
```

Comment:
I _might_ be wrong on this, but is there a need for a GitHub token? If the CI is meant to just check a PR title and error, do we really need a token env variable since it's no like it's going to edit the actual title, it's just going to check if it's correct or not and error

Reply:
https://docs.github.com/en/rest/pulls/pulls?apiVersion=2022-11-28
seems like github require auth tokens (not sure)

Reply:
It's not strictly required in this case, since this is a public repo and we're not using any special features of that action. The endpoint is accessible without auth, but unauthed ratelimits are pretty strict - since the token here only gets `pull_requests: read` perms anyway, I'm not concerned about security that much

Reply:
cool then, seems pr is complete